ste/sender.go

```diff
@@ -153,6 +154,12 @@ func createChunkFunc(setDoneStatusOnExit bool, jptm IJobPartTransferMgr, id comm
 				defer jptm.LogChunkStatus(id, common.EWaitReason.ChunkDone())
 			}
 		}
+
+		// tell the jptm that the destination should be assumed to have been modified
+		// (this is necessary for those cases where the prologue does not modify the dest, so the flag will not have been set at prologue time)
+		// It's idempotent, so we call it every time rather than, say, test OffsetInFile and assume that id.OffsetInFile == 0 will always run first.
+		jptm.SetDestinationIsModified()
```

Comment:
Alright, here's a thought: S2S senders, if the src length is 0 still generate a chunk func. Is this supposed to be set on these as well?

Reply:
Good question. I can see logical reasons for both possible answers (yes and no).  I'm leaning towards "yes, it is correct to set this" because the flag is "destination _modified_" and if we create a zero-length object at the destination then we have indeed "modified" it. (If the flag was called "destination requires cleanup", the argument would be stronger than we should _not_ set it).

BTW, if there's a cancellation in the interval of time between one of those 0-length chunk funcs and the epilogue for that blob, we'll get the block list, and realise we don't need to delete it.  So the result to the user is the same no matter what we set here (no deletion).

So I'm going to leave this code as is, and mark this resolved.